cpp/src/parquet/types.cc

```diff
@@ -463,6 +463,31 @@ std::shared_ptr<const LogicalType> LogicalType::FromThrift(
     return UUIDLogicalType::Make();
   } else if (type.__isset.FLOAT16) {
     return Float16LogicalType::Make();
+  } else if (type.__isset.GEOMETRY) {
+    std::string crs;
+    if (type.GEOMETRY.__isset.crs) {
+      crs = type.GEOMETRY.crs;
+    }
+
+    LogicalType::GeometryEdges::edges edges = LogicalType::GeometryEdges::UNKNOWN;
+    if (type.GEOMETRY.edges == format::Edges::PLANAR) {
+      edges = LogicalType::GeometryEdges::PLANAR;
+    } else if (type.GEOMETRY.edges == format::Edges::SPHERICAL) {
+      edges = LogicalType::GeometryEdges::SPHERICAL;
+    }
+
+    LogicalType::GeometryEncoding::geometry_encoding encoding =
+        LogicalType::GeometryEncoding::UNKNOWN;
+    if (type.GEOMETRY.encoding == format::GeometryEncoding::WKB) {
+      encoding = LogicalType::GeometryEncoding::WKB;
+    }
+
+    std::string metadata;
```

Comment:
The `metadata` field has been changed from string to binary type as suggested by @emkornfield to be more flexible (e.g. encoding): 
 https://github.com/apache/parquet-format/pull/240#discussion_r1639285008. I'm not sure if this makes implementation difficult to deal with the complexity.